src/main/java/racingcar/controller/GameController.java

```diff
@@ -0,0 +1,55 @@
+package racingcar.controller;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import racingcar.domain.car.Cars;
+import racingcar.domain.game.ProgressCount;
+import racingcar.domain.game.RacingGame;
+import racingcar.domain.generator.RacingRandomNumberGenerator;
+import racingcar.view.InputView;
+import racingcar.view.OutputView;
+
+public class GameController {
+
+    public void run() {
+        RacingGame racingGame = createRacingGame();
+        while (racingGame.isRun()) {
+            racingGame.progress();
+            OutputView.printCurrentSituation(racingGame.getCars());
+        }
+        OutputView.printWinner(racingGame.getWinner());
+    }
+
+    private RacingGame createRacingGame() {
+        Cars cars = createCars();
+        ProgressCount progressCount = createProgressCount();
+        return RacingGame.createRacingGame(cars,
+                progressCount);
+    }
+
+    private ProgressCount createProgressCount() {
+        int maxProgressCount = InputView.askMaxProgressCount();
+        return new ProgressCount(maxProgressCount, 0);
+    }
+
+    private Cars createCars() {
+        String str = InputView.askCarNames();
+        validatePattern(str);
```

Comment:
P5: 어떤 Pattern을 validate 하는 것인지 네이밍을 통해 알 수 있으면 좋겠다는 생각이 드네요!

```suggestion
        String carNames = InputView.askCarNames();
        validatePattern(carNames);
```
또한 변수명이 str이라는 자료형의 약자로 사용되면 의미를 파악하기 어렵다고 생각이 듭니다!
때문에 차라리 이렇게 직관적인 네이밍을 사용해보심은 어떠신가요?